sections/main-product.liquid

```diff
@@ -315,52 +322,100 @@
+                    for option1_name in variants_option1_arr
+                      case option.position
+                        when 1
+                          if variants_option1_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+                            assign option_disabled = false
+                          endif
+                        when 2
+                          if option1_name == product.selected_or_first_available_variant.option1 and variants_option2_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+                            assign option_disabled = false
+                          endif
+                        when 3
+                          if option1_name == product.selected_or_first_available_variant.option1 and variants_option2_arr[forloop.index0] == product.selected_or_first_available_variant.option2 and variants_option3_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+                            assign option_disabled = false
+                          endif
+                      endcase
+                    endfor
+                  %}
+
+                  {% capture radio_options_html %}
+                    {{ radio_options_html }}
+                      <input type="radio" id="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}"
+                            name="{{ option.name }}"
+                            value="{{ value | escape }}"
+                            form="{{ product_form_id }}"
+                            {% if option.selected_value == value %}checked{% endif %}
+                            {% if option_disabled %}class="disabled"{% endif %}
+                      >
+                      <label for="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}">
+                        {{ value }}
+                      </label>
+                  {% endcapture %}
+
+                  {% capture select_options_html %}
+                    {{ select_options_html }}
+                    <option value="{{ value | escape }}" {% if option.selected_value == value %}selected="selected"{% endif %}{% if option_disabled and 'disabled' == block.settings.select_soldout_type %} disabled{% endif %}>
+                      {{ value }}{% if option_disabled and 'append' == block.settings.select_soldout_type %} [{{ 'products.product.unavailable' | t }}]{% endif %}
+                    </option>
+                  {% endcapture %}
+
+                {%- endfor -%}
+
+                {% capture radio_options_html %}
+                  {{ radio_options_html }}
+                  </fieldset>
+                {% endcapture %}
+
+                {% capture select_options_html %}
+                  {{ select_options_html }}
+                      </select>
+                      {% render 'icon-caret' %}
+                    </div>
+                  </div>
+                {% endcapture %}
+
+              {%- endfor -%}              
```

Comment:
I think we could actually keep what you had prior to this but just do the variable assignment once. So basically taking out just the logic that sets `option_disabled`.

This I find make me look in a few different place to understand what's outputted where. There is a bit of the HTML structure here but also further down. 

Do you think that'd be doable ?

So maybe if you just kept the original code and added right after the logic you had going: 
```html
{%- when 'variant_picker' -%}
            {%- unless product.has_only_default_variant -%}
```

Reply:
Hmm yeah you're right. I think I prefer the first approach then. More readable even if it's the for loop is there twice. I'm not sure if I can get someone to review the PR right now to get their input but hopefully soon. 

What do you think/what's your take?

Reply:
I agree the original version was easier to read with the multiple loops but I have a brainstorm of a better way if I can get to it today then I'll try.

Reply:
@ludoboludo Moved the `input` and `select` elements into a separate snippet

Reply:
Hey @cfxd sorry I haven't had a chance to re review this yet. Got side tracked by other things. Next week I won't either but I keep it on my list of things to do and get to. 👍

Reply:
Would be great to have this as part of a 6.0.0 reease 🙌

Reply:
@ludoboludo since you need 2 reviewers, maybe someone else can look at it this week? This is more important IMO than most of the other PRs

Reply:
👀 let me try and loop in another dev and designer as well to avoid having to rebase the PR over and over to keep up with the rest.